instance/api/openedx_appserver.py

```diff
@@ -98,5 +98,5 @@ def make_active(self, request, pk):
             return Response(
                 {"error": "Cannot make an unhealthy app server active."}, status=status.HTTP_400_BAD_REQUEST
             )
-        app_server.instance.set_appserver_active(app_server.pk)
-        return Response({'status': 'App server updated.'})
+        set_appserver_active(app_server.pk)
+        return Response({'status': 'App server activation initiated.'})
```

Comment:
Nice language tweak : )

Does the UI correctly refresh once the the server is actually activated? I'm not talking about the feedback from the LB that I was mentioning in the ticket (although, if you have an idea of how to approach that, I would be curious too), but just the UI <-> Django backend part. It seemed to be the case in OpenCraft.Hosting prod when I just tested now, but since I'm not sure which version is live at the moment, I'm double-checking, it seem to have the old "App server updated" copy there.

I have also noticed that some of the haproxy tasks seemed to be failing in the huey logs - I haven't dug though:

```
18:33:24 worker.1    | 11/Nov/2016 18:33:24 |     INFO | process=6584  | instance.models.load_bala | load_balancer=1 (haproxy.opencra) |   Traceback (most recent call last):
18:33:24 worker.1    | 11/Nov/2016 18:33:24 |     INFO | process=6584  | instance.models.load_bala | load_balancer=1 (haproxy.opencra) |     File "<string>", line 1, in <module>
18:33:24 worker.1    | 11/Nov/2016 18:33:24 |     INFO | process=6584  | instance.models.load_bala | load_balancer=1 (haproxy.opencra) |     File "/tmp/tmp2f52a1co/pip-build-_S9Fqw/cryptography/setup.py", line 334, in <module>
18:33:24 worker.1    | 11/Nov/2016 18:33:24 |     INFO | process=6584  | instance.models.load_bala | load_balancer=1 (haproxy.opencra) |       **keywords_with_side_effects(sys.argv)
18:33:25 worker.1    | 11/Nov/2016 18:33:25 |     INFO | process=6584  | instance.models.load_bala | load_balancer=1 (haproxy.opencra) |     File "/usr/lib/python2.7/distutils/core.py", line 111, in setup
18:33:25 worker.1    | 11/Nov/2016 18:33:25 |     INFO | process=6584  | instance.models.load_bala | load_balancer=1 (haproxy.opencra) |       _setup_distribution = dist = klass(attrs)
18:33:25 worker.1    | 11/Nov/2016 18:33:25 |     INFO | process=6584  | instance.models.load_bala | load_balancer=1 (haproxy.opencra) |     File "/tmp/tmp2f52a1co/venv/local/lib/python2.7/site-packages/setuptools/dist.py", line 315, in __init__
18:33:25 worker.1    | 11/Nov/2016 18:33:25 |     INFO | process=6584  | instance.models.load_bala | load_balancer=1 (haproxy.opencra) |       self.fetch_build_eggs(attrs['setup_requires'])
18:33:25 worker.1    | 11/Nov/2016 18:33:25 |     INFO | process=6584  | instance.models.load_bala | load_balancer=1 (haproxy.opencra) |     File "/tmp/tmp2f52a1co/venv/local/lib/python2.7/site-packages/setuptools/dist.py", line 361, in fetch_build_eggs
18:33:25 worker.1    | 11/Nov/2016 18:33:25 |     INFO | process=6584  | instance.models.load_bala | load_balancer=1 (haproxy.opencra) |       replace_conflicting=True,
18:33:25 worker.1    | 11/Nov/2016 18:33:25 |     INFO | process=6584  | instance.models.load_bala | load_balancer=1 (haproxy.opencra) |     File "/tmp/tmp2f52a1co/venv/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 850, in resolve
18:33:25 worker.1    | 11/Nov/2016 18:33:25 |     INFO | process=6584  | instance.models.load_bala | load_balancer=1 (haproxy.opencra) |       dist = best[req.key] = env.best_match(req, ws, installer)
18:33:25 worker.1    | 11/Nov/2016 18:33:25 |     INFO | process=6584  | instance.models.load_bala | load_balancer=1 (haproxy.opencra) |     File "/tmp/tmp2f52a1co/venv/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1122, in best_match
18:33:25 worker.1    | 11/Nov/2016 18:33:25 |     INFO | process=6584  | instance.models.load_bala | load_balancer=1 (haproxy.opencra) |       return self.obtain(req, installer)
18:33:25 worker.1    | 11/Nov/2016 18:33:25 |     INFO | process=6584  | instance.models.load_bala | load_balancer=1 (haproxy.opencra) |     File "/tmp/tmp2f52a1co/venv/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1134, in obtain
18:33:25 worker.1    | 11/Nov/2016 18:33:25 |     INFO | process=6584  | instance.models.load_bala | load_balancer=1 (haproxy.opencra) |       return installer(requirement)
18:33:25 worker.1    | 11/Nov/2016 18:33:25 |     INFO | process=6584  | instance.models.load_bala | load_balancer=1 (haproxy.opencra) |     File "/tmp/tmp2f52a1co/venv/local/lib/python2.7/site-packages/setuptools/dist.py", line 429, in fetch_build_egg
18:33:25 worker.1    | 11/Nov/2016 18:33:25 |     INFO | process=6584  | instance.models.load_bala | load_balancer=1 (haproxy.opencra) |       return cmd.easy_install(req)
18:33:25 worker.1    | 11/Nov/2016 18:33:25 |     INFO | process=6584  | instance.models.load_bala | load_balancer=1 (haproxy.opencra) |     File "/tmp/tmp2f52a1co/venv/local/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 665, in easy_install
18:33:26 worker.1    | 11/Nov/2016 18:33:26 |     INFO | process=6584  | instance.models.load_bala | load_balancer=1 (haproxy.opencra) |       return self.install_item(spec, dist.location, tmpdir, deps)
18:33:26 worker.1    | 11/Nov/2016 18:33:26 |     INFO | process=6584  | instance.models.load_bala | load_balancer=1 (haproxy.opencra) |     File "/tmp/tmp2f52a1co/venv/local/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 695, in install_item
18:33:26 worker.1    | 11/Nov/2016 18:33:26 |     INFO | process=6584  | instance.models.load_bala | load_balancer=1 (haproxy.opencra) |       dists = self.install_eggs(spec, download, tmpdir)
18:33:26 worker.1    | 11/Nov/2016 18:33:26 |     INFO | process=6584  | instance.models.load_bala | load_balancer=1 (haproxy.opencra) |     File "/tmp/tmp2f52a1co/venv/local/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 876, in install_eggs
18:33:26 worker.1    | 11/Nov/2016 18:33:26 |     INFO | process=6584  | instance.models.load_bala | load_balancer=1 (haproxy.opencra) |       return self.build_and_install(setup_script, setup_base)
18:33:26 worker.1    | 11/Nov/2016 18:33:26 |     INFO | process=6584  | instance.models.load_bala | load_balancer=1 (haproxy.opencra) |     File "/tmp/tmp2f52a1co/venv/local/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 1115, in build_and_install
18:33:26 worker.1    | 11/Nov/2016 18:33:26 |     INFO | process=6584  | instance.models.load_bala | load_balancer=1 (haproxy.opencra) |       self.run_setup(setup_script, setup_base, args)
18:33:26 worker.1    | 11/Nov/2016 18:33:26 |     INFO | process=6584  | instance.models.load_bala | load_balancer=1 (haproxy.opencra) |     File "/tmp/tmp2f52a1co/venv/local/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 1103, in run_setup
18:33:26 worker.1    | 11/Nov/2016 18:33:26 |     INFO | process=6584  | instance.models.load_bala | load_balancer=1 (haproxy.opencra) |       raise DistutilsError("Setup script exited with %s" % (v.args[0],))
18:33:26 worker.1    | 11/Nov/2016 18:33:26 |     INFO | process=6584  | instance.models.load_bala | load_balancer=1 (haproxy.opencra) |   distutils.errors.DistutilsError: Setup script exited with error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
18:33:26 worker.1    | 11/Nov/2016 18:33:26 |     INFO | process=6584  | instance.models.load_bala | load_balancer=1 (haproxy.opencra) |
18:33:26 worker.1    | 11/Nov/2016 18:33:26 |     INFO | process=6584  | instance.models.load_bala | load_balancer=1 (haproxy.opencra) |   ----------------------------------------
18:33:26 worker.1    | 11/Nov/2016 18:33:26 |     INFO | process=6584  | instance.models.load_bala | load_balancer=1 (haproxy.opencra) |   Running setup.py clean for cryptography
18:33:26 worker.1    | 11/Nov/2016 18:33:26 |     INFO | process=6584  | instance.models.load_bala | load_balancer=1 (haproxy.opencra) |   Complete output from command /tmp/tmp2f52a1co/venv/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/tmp2f52a1co/pip-build-_S9Fqw/cryptography/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" clean --all:
```

I tried to go to the instance log in the UI, but I couldn't find the logs of the ansible task run there: https://console.opencraft.com/instance/1890/edx-appserver/601/ - are we attaching these logs to their instance?